### PR TITLE
Add algorithms for link-based trees

### PR DESCRIPTION
**Pull request by bifurcation**
As discussed in #507, having algorithms for a "link-based" representation of the tree would help emphasize that the array-based tree representation is not required.  This PR adds them in an appendix.

### CODE REVIEW

**Comment by bifurcation** [comment]
cc @TWal @franziskuskiefer

File: draft-ietf-mls-protocol.md [hunk 1]
```diff
@@ -4092,3 +4091,118 @@ def common_ancestor_direct(x, y, _):
...
+the array.)  Such an implementation needs to update these links to maintain the
+left-balanced structure of the tree as the tree is extended to add new members,
+or truncated when memebers are removed.
+
+This appendix provides example algorithms for adding a single leaf and removing
+a single leaf from a link-based tree.  These algorithms can be repeated as
+necessary to implement the tree modifications required to apply Add and Remove
+proposals.
+
+To add a leaf node N to the right side of the tree:
```
Comment (franziskuskiefer) [inline draft-ietf-mls-protocol.md:4111]: Do we need both, pseudo code and python code? I'd be fine with the python code only and some of this as comments if you feel like it.

**Review by franziskuskiefer — COMMENTED** [review]
lgtm

**Comment by bifurcation** [comment]
Subsumed by #522

**Pull request closed without merging**